core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -438,6 +449,18 @@ public synchronized void reset() throws IOException {
       count = mark;
     }
 
+    @Override
+    public void close() throws IOException {
+      super.close();
+    }
+
+    private void reportCount() {
+      if (count > maxCount) {
+        statsTraceContext.messageReceived(count - maxCount);
```

Comment:
The reader of the InputStream is the Marshaller. How the reads are segmented depends on Marshaller implementation, which is not a useful information to be exposed. Marking the arrival of each gRPC message makes more sense.